accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,242 @@
+- Add a new entry to table `susesccrepositoryauth` for each repository, where:
+  - repo_id: the one obtained in previous step
+  - credentials_id: the one obtain from `susecredentials`
+  - source_id: null
+  - auth_type: `cloudrmt`
+
+We also need to save the repository base URL in `suseCredentials` table. URL from table `susesccrepository` is pointing to SCC and we need to connect to the RMT cloud servers. For that we need to save URL prefix, including hostname.
+Example of RMT base URL:
+- URL on pay-as-you-go instance: `https://host/repo/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+- Sub-string used to search repository on `susSccrespository` table: `/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+- Repository base URL to save on `suseCredentials` table: `https://host/repo`
+
+## Add product
+
+When a product is added the table `rhncontentsource` is populated according to the authentication mechanism defined in table `susesccrepositoryauth`. The url saved to `rhncontentsource` depends on the authentication method in use.
+
+For the new `cloudrmt` authentication method the url should be composed as:
+- Concatenation of base URL from `suseCredentials` with the url path from `susesccrepository`
+- Authentication will use the same mechanism as basic authentication, where id of `suseCredentials` is passed as query string in the repository URL
+
+## Teach reposync on how to use Cloud RMT authentication mechanism
+
+Reposync is already able to deal with the basic authentication mechanism. It receives the id of a `suseCredentials` record and loads the basic authentication data.
+This mechanism will be enhanced to consider also the new column with the authentication headers. The following changes need to be implemented:
+  - Modify method [`_url_with_repo_credentials`](https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/reposync.py#L1785) to also load and return the http authentication headers from table `suseCredentials`
+  - Enhance [repository plugins](https://github.com/uyuni-project/uyuni/tree/master/backend/satellite_tools/repo_plugins) to receive http headers
+  - Enhance reposync [zypper plugin](https://github.com/uyuni-project/uyuni/blob/master/backend/satellite_tools/spacewalk-extra-http-headers) to receive http header. Header will be written to a temporary file in the same location of zypper `.repo` file.
+
+Another mechanism for [authentication headers](https://github.com/uyuni-project/uyuni/pull/2136) is already in place which loads headers from a configuration file. We should cope with this implementation and merger header loaded form both locations.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Uyuni/SUSE manager pay-as-you-go
+
+We could define a uyuni/suse manager pay-as-you-go image with access to all cloud RMT repositories.
+It would be possible to synchronize any product directly from cloud in a more simple and straightforward way.
```

Comment:
"cloud", "Cloud" and "Public Cloud" are used somewhat interchangeably throughout the document. Please settle on one piece of jargon, if necessary define it  explicitly, and use it consistently.

Reply:
I have change the text to be consistent and always refer "public cloud"